core/domain/collection_services.py

```diff
@@ -526,6 +531,14 @@ def _save_collection(committer_id, collection, commit_message, change_list):
     else:
         collection.validate(strict=False)
 
+    for collection_node in collection.nodes:
+        if not exp_services.get_exploration_by_id(
+            collection_node.exploration_id, strict=False):
+            raise Exception(
```

Comment:
Could you throw a ``utils.ValidationError``, instead? Also, consider the following string as a message instead, since it better matches the style of validation errors using in domain objects:

```python
'Expected collection to only reference valid explorations, but found an exploration with ID: %s'
```